### PR TITLE
[REA] Add optional entries to simple format

### PR DESCRIPTION
**Pull request by Leo40Git**
This PR adds the ability to declare entries as optional in the in object-based attachment dictionaries, like so:
```json5
{
  "replace": false,
  "values": {
    "aurorasdeco:azaela_log?": false, // this entry is optional!
    "#minecraft:planks?": true        // prints a warning - tag entries are always optional, this is redundant...
  }
}
```
Additionally, this PR fixes a potential bug where the reloader attempts to associate values with nonexistent registry entries.

### CODE REVIEW

**Comment by Leo40Git** [comment]
oops, didn't mean to open this as a draft

File: library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/reloader/AttachmentDictionary.java [hunk 1]
```diff
@@ -151,88 +151,86 @@ private void handleArray(Identifier resourceId, Resource resource, JsonArray val
...
-			if (parsedValue == null) {
-				continue;
-			}
-
-			if (tagId) {
-				this.putTag(id, parsedValue);
-			} else {
-				this.put(id, parsedValue);
-			}
+			handleEntry(resourceId, id, isTag, required, value);
```
Comment (LambdAurora) [inline library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/reloader/AttachmentDictionary.java]:
```suggestion
			this.handleEntry(resourceId, id, isTag, required, value);
```

File: library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/reloader/AttachmentDictionary.java [hunk 2]
```diff
@@ -151,88 +151,86 @@ private void handleArray(Identifier resourceId, Resource resource, JsonArray val
...
 						resourceId, entry.getKey());
 				LOGGER.error("", e);
 				continue;
 			}
 
-			if (!tagId && !registry.containsId(id)) {
-				LOGGER.error("Unregistered identifier in values of {}: '{}', ignoring", resourceId, id);
-				continue;
-			}
+			handleEntry(resourceId, id, isTag, required, entry.getValue());
```
Comment (LambdAurora) [inline library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/reloader/AttachmentDictionary.java]:
```suggestion
			this.handleEntry(resourceId, id, isTag, required, entry.getValue());
```

**Review by LambdAurora — COMMENTED** [review]
Yes, I'm gonna be annoying about the `this` keyword again :P

**Review by OroArmor — APPROVED** [review]

**Review by EnnuiL — APPROVED** [review]
You're my lord and savior; PR approved!

**Review by oliviathevampire — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

**Pull request merged**